core/templates/dev/head/services/contextual/UrlServiceSpec.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the BackgroundMaskService.
+ */
+
+describe('Url Service', function() {
+  var UrlService = null;
+  var parameterList = '?parent=parent1&parent=parent2';
+  var location = {
+    href: 'http://sample.com' + parameterList,
+    search: parameterList
+  };
+
+  beforeEach(module('oppia'));
+  beforeEach(inject(function($injector) {
+    UrlService = $injector.get('UrlService');
+    spyOn(UrlService, 'getCurrentUrl').and.returnValue(location);
+  }));
+
+  it('should get correct parameters', function() {
```

Comment:
In general, for all these tests:
- try to get UrlService.js to 100% test coverage
- add more tests beyond that to test corner cases, e.g. no params, multiple different keys, etc.

Reply:
Done